juriscraper/pacer/http.py

```diff
@@ -228,164 +232,86 @@ def _get_xml_view_state(r):
-            raise PacerLoginException(
-                "Password must be at least eight " "characters"
-            )
-        if "timeout error" in login_post_r.text:
-            raise PacerLoginException("Timeout")
 
-        if not self.cookies.get("PacerSession"):
-            logger.info(
-                "Did not get cookies from first log in POSTs. "
-                "Assuming this is a filing user and doing two more "
-                "POSTs."
-            )
-            reg_msg_r = self.post(
-                url,
-                headers={"User-Agent": "Juriscraper"},
-                verify=False,
-                timeout=60,
-                auto_login=False,
-                data={
-                    "javax.faces.partial.ajax": "true",
-                    "javax.faces.source": "regmsg:chkRedact",
-                    "javax.faces.partial.execute": "regmsg:chkRedact",
-                    "javax.faces.partial.render": "regmsg:bpmConfirm",
-                    "javax.faces.partial.event": "change",
-                    "javax.faces.behavior.event": "valueChange",
-                    "javax.faces.ViewState": self._get_xml_view_state(
-                        login_post_r
-                    ),
-                    "regmsg": "regmsg",
-                    "regmsg:chkRedact_input": "on",
-                },
-            )
-            # The box is now checked. Submit the form to say so.
-            self.post(
-                url,
-                headers={"User-Agent": "Juriscraper"},
-                verify=False,
-                timeout=60,
-                auto_login=False,
-                data={
-                    "javax.faces.partial.ajax": "true",
-                    "javax.faces.source": "regmsg:bpmConfirm",
-                    "javax.faces.partial.execute": "@all",
-                    "javax.faces.ViewState": self._get_xml_view_state(
-                        reg_msg_r
-                    ),
-                    "regmsg": "regmsg",
-                    "regmsg:chkRedact_input": "on",
-                    "regmsg:bpmConfirm": "regmsg:bpmConfirm",
-                    "dialogName": "redactionDlg",
-                },
-            )
-
-        if not self.cookies.get("PacerSession") and not self.cookies.get(
-            "NextGenCSO"
+        response_json = login_post_r.json()
+        # 'loginResult': '0', user successfully logged; '1', user not logged
+        if (
+            response_json.get("loginResult") == None
+            or response_json.get("loginResult") == "1"
+        ):
+            message = f"Invalid username/password: {response_json.get('errorDescription')}"
+            raise PacerLoginException(message)
+        # User logged, but with pending actions for their account
+        if (
+            response_json.get("loginResult") == "0"
+            and response_json.get("errorDescription") != None
+            and response_json.get("errorDescription") != ""
```

Comment:
This would be a nice simplification, right?

```suggestion
            and response_json.get("errorDescription")
```